langchain/chains/fork_with_follow_ups.py

```diff
@@ -0,0 +1,54 @@
+"""Conditionally executes a follow up chain based on the output of a decision chain."""
+from typing import Dict, List
+
+from pydantic import BaseModel, Extra, validator
+
+from langchain.chains import LLMChain
+from langchain.chains.base import Chain
+
+
+class ForkChain(Chain, BaseModel):
+    """Conditionally executes follow up chain based on output of a decision chain."""
+
+    decision_chain: LLMChain
+    follow_up_chains: Dict[str, Chain]
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    @validator("follow_up_chains")
+    def default_in_follow_up_chains(cls, v: Dict[str, Chain]) -> Dict[str, Chain]:
```

Comment:
Perhaps it would be better to use defaultdict

https://docs.python.org/3/library/collections.html#collections.defaultdict
https://docs.python.org/3/library/typing.html